apps/web/modules/ee/contacts/api/client/[environmentId]/contacts/[userId]/attributes/lib/contact.ts

```diff
@@ -0,0 +1,41 @@
+import { contactCache } from "@/lib/cache/contact";
+import { cache as reactCache } from "react";
+import { prisma } from "@formbricks/database";
+import { cache } from "@formbricks/lib/cache";
+
+export const getContactByUserIdWithAttributes = reactCache(
+  (environmentId: string, userId: string, updatedAttributes: Record<string, string>) =>
+    cache(
+      async () => {
+        const contact = await prisma.contact.findFirst({
+          where: {
+            environmentId,
+            attributes: { some: { attributeKey: { key: "userId", environmentId }, value: userId } },
+          },
+          select: {
+            id: true,
+            attributes: {
+              where: {
+                attributeKey: {
+                  key: {
+                    in: Object.keys(updatedAttributes),
+                  },
+                },
+              },
+              select: { attributeKey: { select: { key: true } }, value: true },
+            },
+          },
+        });
+
+        if (!contact) {
+          return null;
+        }
+
+        return contact;
+      },
+      [`getContactByUserId-${environmentId}-${userId}-${JSON.stringify(updatedAttributes)}`],
+      {
+        tags: [contactCache.tag.byEnvironmentIdAndUserId(environmentId, userId)],
```

Comment:
we also need to make this dependent on the attributes cache for the user and attributeKey table for the environment